### PR TITLE
Handle minikube de-activate

### PR DESCRIPTION
**Pull request by jakolehm**
Also upgrades extension api to 4.0.0-rc.1.

### CODE REVIEW

**Review by jim-docker — APPROVED** [review]
While this extension is enabled, if I remove the minikube cluster from the minikube workspace it reappears within 5s, is this intended?

Also, there are some problems in the Lens UI with the minikube workspace as added from this extension, logged issue https://github.com/lensapp/lens/issues/1556

**Comment by jakolehm** [comment]
> While this extension is enabled, if I remove the minikube cluster from the minikube workspace it reappears within 5s, is this intended?

This PR should fix that reappear 🤔

**Comment by Nokel81** [comment]
Given that this PR forces the user to change to the minikube cluster when switching to this workspace, shouldn't it disable it or something when going to a new workspace? There was https://github.com/lensapp/lens/pull/1444 but that doesn't seem to be working if there are no other clusters (for instance, default is empty). Maybe that is just a bug with that PR...

**Comment by jim-docker** [comment]
> > While this extension is enabled, if I remove the minikube cluster from the minikube workspace it reappears within 5s, is this intended?
> 
> This PR should fix that reappear 🤔

It fixes it when the extension is deactiviated, but when the extension is active/enabled it still adds it back after I remove it. But this is probably the intent of the extension(?) Not just to add the minkube cluster to the minkube workspace, but also ensure it is always there? It is a bit odd but it does demonstrate that an extension can be active with the workspace and cluster stores while Lens is running.

**Comment by Nokel81** [comment]
So after talking with Jari, it shouldn't be possible to remove a "manged" cluster. Namely one that has a lens' owner (such as in this extension). That remove field shouldn't be present.

File: minikube-sample/main.ts [hunk 1]
```diff
@@ -8,12 +8,35 @@ const workspaceStore = Store.workspaceStore
...
+      clearTimeout(this.syncTimer);
+    }
+
+    const workspace = workspaceStore.getByName("minikube");
+    if (!workspace) {
+      return;
+    }
+
+    if (workspaceStore.currentWorkspaceId === workspace.id) {
+      workspaceStore.setActive(workspaceStore.enabledWorkspacesList[0].id);
```
Comment (Nokel81) [inline minikube-sample/main.ts:32]: Shouldn't it just set it to `WorkspaceStore.defaultId`?

File: minikube-sample/main.ts [hunk 2]
```diff
@@ -8,12 +8,35 @@ const workspaceStore = Store.workspaceStore
...
 
     this.syncMinikube()
   }
 
+  async onDeactivate(): Promise<void> {
+    console.log("minikube extension deactivated")
+
+    if (this.syncTimer) {
+      clearTimeout(this.syncTimer);
+    }
```
Comment (Nokel81) [inline minikube-sample/main.ts:24]: `clearTimeout` already handles this case.
```suggestion
    clearTimeout(this.syncTimer);
```

**Comment by jim-docker** [comment]
> So after talking with Jari, it shouldn't be possible to remove a "manged" cluster. Namely one that has a lens' owner (such as in this extension). That remove field shouldn't be present.

That works, but then I'm still concerned about orphaned workspaces/clusters (https://github.com/lensapp/lens/pull/1573#issuecomment-735919735)

**Comment by Nokel81** [comment]
It is possible to add some functionality to handle that case, either marking them as no longer owned, or removing them.

I think that, if possible, we should have `ownerRef` be required to be the same as the extension that added it. But it might be a bit late for that.

**Pull request closed without merging**